frontend/src/components/popup/Settings.tsx

```diff
@@ -228,6 +230,69 @@ const SectionConflictSettings = memo(function SectionConflictSettings() {
     );
 });
 
+const ExperimentalFeaturesSettings = memo(
+    function ExperimentalFeaturesSettings() {
+        const experimentalFeaturesOptions = useStore(
+            (store) => store.experimentalFeaturesOptions,
+        );
+        const setExperimentalFeaturesOptions = useStore(
+            (store) => store.setExperimentalFeaturesOptions,
+        );
+        const rangeOptions = [2, 4, 6, 8, 10, "all"];
+
+        return (
+            <div className={Css.experimentalFeatures}>
+                <h3 className={Css.title}>Experimental Features</h3>
+                <span>
+                    Search for matching courses from recent terms when no course
+                    is found
+                </span>
+
+                <Slider
+                    value={experimentalFeaturesOptions.enableHistoricalSearch}
+                    onToggle={() => {
+                        setExperimentalFeaturesOptions({
+                            ...experimentalFeaturesOptions,
+                            enableHistoricalSearch:
+                                !experimentalFeaturesOptions.enableHistoricalSearch,
+                        });
+                    }}
+                    text=""
+                />
+                <span>
+                    How many recent terms do you want to search?{" "}
+                    <span className={Css.warning}>
+                        {" "}
+                        (this may affect performance!)
+                    </span>
+                </span>
+                <Dropdown
+                    choices={rangeOptions}
+                    selected={
+                        experimentalFeaturesOptions.historicalSearchRange !==
+                        Infinity
```

Comment:
You should probably use `null` or `undefined` instead of `Infinity` because it can cause some very slippery bug if you accidentally did arithmetic on it (e.g. `Infinity + 1 === Infinity`). If you use `null` the type checker would catch it if you accidentally did math on it.

Reply:
I used `Infinity` as an encoding for the case where the user want to search through "all" historical terms. I exclusively used this it with comparison such that we would keep the `min` between the user's desired range (2, 4, ..., 10, `Infinity`) and how many terms we have because this will ensure that we would never look back more than the terms we have in the database, and keep the case where the user want to search through all terms simply represented `infinity` range. If this causes problem, I could try encode the "all" case the other way.

Reply:
You don't need to guess how many terms we have in the database. You can get a list of all of them with `useAllTerms()` in `hooks/terms.ts`

Reply:
I guess what I'm saying is, the list of options should be dynamically computed based on the number of past semesters available instead of a hard-coded list of 2,4...10, all. For example, the choice can be all the powers of 2 underneath the max (including 1), plus the max itself as a number.